crates/sui-framework/sources/math.move

```diff
@@ -64,4 +64,47 @@ module sui::math {
 
         (res as u64)
     }
+
+    /// Similar to math::sqrt, but for u128 numbers. Get a nearest lower integer Square Root for `x`. Given that this
+    /// function can only operate with integers, it is impossible
+    /// to get perfect (or precise) integer square root for some numbers.
+    ///
+    /// Example:
+    /// ```
+    /// math::sqrt_u128(9) => 3
+    /// math::sqrt_u128(8) => 2 // the nearest lower square root is 4;
+    /// ```
+    ///
+    /// In integer math, one of the possible ways to get results with more
+    /// precision is to use higher values or temporarily multiply the
+    /// value by some bigger number. Ideally if this is a square of 10 or 100.
+    ///
+    /// Example:
+    /// ```
+    /// math::sqrt_u128(8) => 2;
+    /// math::sqrt_u128(8 * 10000) => 282;
+    /// // now we can use this value as if it was 2.82;
+    /// // but to get the actual result, this value needs
+    /// // to be divided by 100 (because sqrt_u128(10000)).
+    ///
+    ///
+    /// math::sqrt_u128(8 * 1000000) => 2828; // same as above, 2828 / 1000 (2.828)
+    /// ```
+    public fun sqrt_u128(x: u128): u128 {
```

Comment:
Per the offline discussion -- we should go down the route of a different math module per integer width.  This can be done in a different PR though (probably it makes sense to split this change out anyway).

Reply:
Yes I think separation is a diff PR. This case here was just added to illustrate usage of u256.